src/main/java/io/anserini/rerank/AxiomReranker.java

```diff
@@ -0,0 +1,302 @@
+        if (filterPattern == null || filterPattern.matcher(term).matches()) {
+          if (!termDocidSets.containsKey(term)) {
+            termDocidSets.put(term, new HashSet<>());
+          }
+          termDocidSets.get(term).add(docid);
+        }
+      }
+    }
+    return termDocidSets;
+  }
+
+  /** Calculate the scores (weights) of each term that occured in the reranking pool.
+   *  The Process:
+   *  1. For each query term, calculate its score for each term in the reranking pool. the score
+   *  is calcuated as P(both occurs)*log{P(both occurs)/P(t1 occurs)/P(t2 occurs)}
+   *  + P(both not occurs)*log{P(both not occurs)/P(t1 not occurs)/P(t2 not occurs)}
+   *  + P(t1 occurs t2 not occurs)*log{P(t1 occurs t2 not occurs)/P(t1 occurs)/P(t2 not occurs)}
+   *  + P(t1 not occurs t2 occurs)*log{P(t1 not occurs t2 occurs)/P(t1 not occurs)/P(t2 occurs)}
+   *  2. For each query term the scores of every other term in the reranking pool are stored in a
+   *  PriorityQueue, only the top L are kept.
+   *  3. Add the scores of the same term together and pick the top K ones.
+   *
+   *  @param termInvertedList    A Map of <term -> Set<docId>> where the Set of docIds is where the term occurs
+   *  @param context             An instance of RerankerContext
+   *
+   *  @return Map<String, Double> Top terms and their weight scores in a HashMap
+   */
+  @VisibleForTesting
+  private Map<String, Double> calTermScore(
+      Map<String, Set<Integer>> termInvertedList, RerankerContext context) {
+    class ScoreComparator implements Comparator<Pair<String, Double>> {
+      public int compare(Pair<String, Double> a, Pair<String, Double> b) {
+        return Double.compare(b.getRight(), a.getRight());
+      }
+    }
+
+    //calculate the Mutual Information between term with each query term
+    List<String> queryTerms = context.getQueryTokens();
+    Map<String, Integer> queryTermsCounts = new HashMap<>();
+    for (String qt : queryTerms) {
+      queryTermsCounts.put(qt, queryTermsCounts.getOrDefault(qt, 0) + 1);
+    }
+
+    Set<Integer> allDocIds = new HashSet<>();
+    for (Set<Integer> s : termInvertedList.values()) {
+      allDocIds.addAll(s);
+    }
+    int docIdsCount = allDocIds.size();
+
+    List<PriorityQueue<Pair<String, Double>>> allTermScoresPQ = new ArrayList<>();
+    for (Map.Entry<String, Integer> q : queryTermsCounts.entrySet()) {
+      String queryTerm = q.getKey();
+      int qtf = q.getValue();
+      if (termInvertedList.containsKey(queryTerm)) {
+        PriorityQueue<Pair<String, Double>> termScorePQ = new PriorityQueue<>(new ScoreComparator());
+        double selfMI = calMutualInformation( termInvertedList.get(queryTerm), termInvertedList.get(queryTerm), docIdsCount );
+        for (Map.Entry<String, Set<Integer>> termEntry : termInvertedList.entrySet()) {
+          double score = 1.0 * qtf;
+          if (!termEntry.getKey().equals(queryTerm)) { // do not include the query term itself ...
```

Comment:
I think it's cleaner if you rewrite with `if (termEntry.getKey().equals(queryTerm)) continue` - cuts down on the indentation level.